arch/arm/core/aarch32/cortex_m/fault.c

```diff
@@ -98,6 +98,16 @@ LOG_MODULE_DECLARE(os, CONFIG_KERNEL_LOG_LEVEL);
  */
 #define ADDITIONAL_STATE_CONTEXT_WORDS 10
 
+#if defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
+/* utility macros for accessing MEMFAULT/BUSFAULT/USGFAULT registers */
+#define SCB_CFSR_MEMFAULTSR_GET() \
+	((SCB->CFSR >> SCB_CFSR_MEMFAULTSR_Pos) & SCB_CFSR_MEMFAULTSR_Msk)
+#define SCB_CFSR_BUSFAULTSR_GET() \
+	((SCB->CFSR >> SCB_CFSR_BUSFAULTSR_Pos) & SCB_CFSR_BUSFAULTSR_Msk)
+#define SCB_CFSR_USGFAULTSR_GET() \
+	((SCB->CFSR >> SCB_CFSR_USGFAULTSR_Pos) & SCB_CFSR_USGFAULTSR_Msk)
+#endif /* CONFIG_ARMV7_M_ARMV8_M_MAINLINE */
```

Comment:
This does not seem right.

https://github.com/ARM-software/CMSIS_5/blob/2e1a5d13866f7318aa5905b7fc8de69382cbd804/CMSIS/Core/Include/core_cm4.h#L616-L623

Reply:
Should have spotted that 🫣. Thanks for catching that @stephanosio